src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/PartyFlagsMappingTest.java

```diff
@@ -381,4 +384,23 @@ void adjournCaseInterpreterLanguageNull(String interpreterLanguage) {
 
         assertThat(result).isNull();
     }
+
+    @DisplayName("get Case Flags")
+    @Test
+    void shouldGetCaseFlags() {
+        // given
+        SscsCaseData sscsCaseData = Mockito.mock(SscsCaseData.class);
+        Appeal appeal = Mockito.mock(Appeal.class);
+        HearingOptions hearingOptions = Mockito.mock(HearingOptions.class);
+        // when
+        Mockito.when(hearingOptions.getSignLanguageType()).thenReturn("British Sign Language (BSL)");
+        Mockito.when(appeal.getHearingOptions()).thenReturn(hearingOptions);
+        Mockito.when(sscsCaseData.getAppeal()).thenReturn(appeal);
+        // then
+        CaseFlags caseFlags = PartyFlagsMapping.getCaseFlags(sscsCaseData);
+        assertEquals("", caseFlags.getFlagAmendUrl());
+        assertEquals(1, caseFlags.getFlags().size());
+        assertEquals(SIGN_LANGUAGE_TYPE.getFlagId(), caseFlags.getFlags().stream().findFirst().orElseThrow().getFlagId());
```

Comment:
why orElseThrow() ? we just use get() - the call should be okay as you are making the assertion above - Sonar might complain though..